features/admin.actions.v1/components/action-config-form.tsx

```diff
@@ -0,0 +1,841 @@
+import { AxiosError, AxiosResponse } from "axios";
+import React, { Fragment, FunctionComponent, ReactElement, useEffect, useState } from "react";
+import { Trans, useTranslation } from "react-i18next";
+import { useDispatch, useSelector } from "react-redux";
+import { Dispatch } from "redux";
+
+import { Icon } from "semantic-ui-react";
+import { KeyedMutator } from "swr";
+import createAction from "../api/create-action";
+import updateAction from "../api/update-action";
+import { ActionsConstants } from "../constants/actions-constants";
+import {
+    ActionConfigFormPropertyInterface,
+    ActionInterface,
+    ActionResponseInterface,
+    ActionUpdateInterface,
+    AuthenticationPropertiesInterface,
+    AuthenticationType,
+    AuthenticationTypeDropdownOption
+} from "../models/actions";
+import "./action-config-form.scss";
+
+/**
+ * Prop types for the action configuration form component.
+ */
+interface ActionConfigFormInterface extends IdentifiableComponentInterface, LoadableComponentInterface {
+    /**
+     * Action's initial values.
+     */
+    initialValues: ActionConfigFormPropertyInterface;
+    /**
+     * Action Type of the Action.
+     */
+    actionTypeApiPath: string;
+    /**
+     * Specifies action creation state.
+     */
+    isCreateFormState: boolean;
+    /**
+     * Mutate api.
+     */
+    mutate: KeyedMutator<AxiosResponse<ActionResponseInterface[]>>
+}
+
+export const ActionConfigForm: FunctionComponent<ActionConfigFormInterface> = ({
+    initialValues,
+    isLoading,
+    actionTypeApiPath,
+    isCreateFormState,
+    mutate: mutateActions,
+    "data-componentid": _componentId = "action-config-form"
+}: ActionConfigFormInterface): ReactElement => {
+
+    const featureConfig: FeatureConfigInterface = useSelector((state: AppState) => state.config.ui.features.actions);
+    const [ isAuthenticationUpdateFormState, setIsAuthenticationUpdateFormState ] = useState<boolean>(false);
+    const [ authenticationType, setAuthenticationType ] = useState<AuthenticationType>(null);
+    const [ isSubmitting, setIsSubmitting ] = useState(false);
+    const [ isShowSecret1, setIsShowSecret1 ] = useState(false);
+    const [ isShowSecret2, setIsShowSecret2 ] = useState(false);
+
+    const dispatch: Dispatch = useDispatch();
+    const { t } = useTranslation();
+
+    const hasActionUpdatePermissions: boolean = useRequiredScopes(featureConfig?.actions?.scopes?.update);
+    const hasActionCreatePermissions: boolean = useRequiredScopes(featureConfig?.actions?.scopes?.create);
+
+    /**
+     * The following useEffect is used to set the current Action Authentication Type.
+     */
+    useEffect(() => {
+        if (initialValues?.id === undefined) {
+            setIsAuthenticationUpdateFormState(true);
+        } else {
+            setAuthenticationType(initialValues.authenticationType as AuthenticationType);
+            setIsAuthenticationUpdateFormState(false);
```

Comment:
Can't we replace this with `handleAuthenticationChangeCancel`

Reply:
`handleAuthenticationChangeCancel` is for handling authentication section change. 
`isAuthenticationUpdateFormState ` is for handling whole form update state